lib/Sema/DerivedConformanceParameterAggregate.cpp

```diff
@@ -73,6 +73,27 @@ bool DerivedConformance::canDeriveParameterAggregate(TypeChecker &TC,
   return bool(deriveParameterAggregate_Parameter(TC, nominal));
 }
 
+// Add @_fixed_layout attribute to type conforming to `ParameterAggregate`, if
+// necessary. @_fixed_layout is currently necessary for differentiation wrt
+// structs conforming to `ParameterAggregate` for resilience reasons.
+void addFixedLayoutAttrIfNeeded(TypeChecker &TC, NominalTypeDecl *nominal) {
+  // If nominal already has @_fixed_layout, return.
+  if (nominal->getAttrs().hasAttribute<FixedLayoutAttr>()) return;
+  auto access = nominal->getEffectiveAccess();
+  // If nominal does not have at least internal access, return.
+  if (access < AccessLevel::Internal)
+    return;
+  // If nominal is internal, it should have the @usableFromInline attribute.
+  if (access == AccessLevel::Internal &&
+      nominal->getAttrs().hasAttribute<UsableFromInlineAttr>()) {
```

Comment:
I don't think the condition is correct. This adds a `@usableFromInline` when the type already has `@usableFromInline`. Did you mean to negate the second condition?

Reply:
Thanks for the catch. Fixed in https://github.com/apple/swift/pull/18507/commits/aa37d55cd2ef66a0a9e8f92278aa62c8406fbe08.